lib/query-interface.js

```diff
@@ -214,16 +213,12 @@ class QueryInterface {
     );
 
     // Postgres requires special SQL commands for ENUM/ENUM[]
-    if (this.sequelize.options.dialect === 'postgres') {
-      promise = PostgresQueryInterface.ensureEnums(this, tableName, attributes, options, model);
-    } else {
-      promise = Promise.resolve();
-    }
+    const promise = this.sequelize.options.dialect === 'postgres' ?
+      PostgresQueryInterface.ensureEnums(this, tableName, attributes, options, model) :
+      Promise.resolve();
 
-    if (
-      !tableName.schema &&
-      (options.schema || !!model && model._schema)
-    ) {
+    const canAddSchema = !tableName.schema && (options.schema || !!model && model._schema);
```

Comment:
The `!!` boolean cast is unnecessary, also I know it's debatable but I think adding parentheses around the `&&` improves readability; not everyone knows that `&&` has higher precedence than `||` in JS.

```suggestion
    const canAddSchema = !tableName.schema && (options.schema || (model && model._schema));
```